File: tools/java/FileProbe.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public final class FileProbe {
    private FileProbe() {
    }

    private static String jsonEscape(String value) {
        return value
            .replace("\\", "\\\\")
            .replace("\"", "\\\"")
            .replace("\n", "\\n")
            .replace("\r", "\\r")
            .replace("\t", "\\t");
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("usage: java FileProbe <file>");
            System.exit(2);
        }

        Path path = Path.of(args[0]);
        try {
            long size = Files.size(path);
            String name = path.getFileName() == null ? "" : path.getFileName().toString();
            System.out.printf(
                "{\"language\":\"java\",\"name\":\"%s\",\"size\":%d}%n",
                jsonEscape(name),
                size
            );
        } catch (IOException error) {
            System.err.println(error.getMessage());
            System.exit(1);
        }
    }
}

File: tools/rust/file_probe.rs
use std::env;
use std::fs;
use std::path::Path;

fn json_escape(value: &str) -> String {
    value
        .replace('\\', "\\\\")
        .replace('"', "\\\"")
        .replace('\n', "\\n")
        .replace('\r', "\\r")
        .replace('\t', "\\t")
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 2 {
        eprintln!("usage: fileprobe-rust <file>");
        std::process::exit(2);
    }

    let path = Path::new(&args[1]);
    let metadata = match fs::metadata(path) {
        Ok(value) => value,
        Err(error) => {
            eprintln!("{}", error);
            std::process::exit(1);
        }
    };
    let name = path.file_name().and_then(|value| value.to_str()).unwrap_or("");
    println!(
        "{{\"language\":\"rust\",\"name\":\"{}\",\"size\":{}}}",
        json_escape(name),
        metadata.len()
    );
}
